src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/FlowCrudV2Spec.groovy

```diff
@@ -1104,6 +1103,155 @@ class FlowCrudV2Spec extends HealthCheckSpecification {
         }
     }
 
+    @Tidy
+    def "System doesn't reroute flow to more preferable path while updating portNumber/vlanId"() {
+        given: "Two active switches connected to traffgens with two possible paths at least"
+        def allTraffgenSwitchIds = topology.activeTraffGens*.switchConnected*.dpId ?:
+                assumeTrue("Should be at least two active traffgens connected to switches", false)
+        def switchPair = topologyHelper.getAllNeighboringSwitchPairs().find { swP ->
+            allTraffgenSwitchIds.contains(swP.src.dpId) && allTraffgenSwitchIds.contains(swP.dst.dpId) &&
+                    swP.paths.size() >= 2
+        } ?: assumeTrue("Unable to find required switches/paths in topology",false)
+
+        and: "A flow"
+        def flow = flowHelperV2.randomFlow(switchPair, false)
+        flowHelperV2.addFlow(flow)
+
+        when: "Make the current path less preferable than alternatives"
+        def currentPath = pathHelper.convert(northbound.getFlowPath(flow.flowId))
+        def alternativePaths = switchPair.paths.findAll { it != currentPath }
+        alternativePaths.each { pathHelper.makePathMorePreferable(it, currentPath) }
+
+        and: "Update the flow: port number and vlanId on the src/dst endpoints"
+        def updatedFlow = flow.jacksonCopy().tap {
```

Comment:
it will be better if you verify separately all three scenarios:
- src endpoint changed
- dst endpoint changed
- both endpoints changed

Reply:
added